integration/README.md

```diff
@@ -0,0 +1,48 @@
+# Setup the FPC test network
+
+The FPC test network builds on the test-network provided by fabric-samples.
+
+We provide fabric-samples as a submodule in `$FPC_PATH/test-network/fabric-samples`.
+
+Make sure you have installed [yq](https://github.com/mikefarah/yq).
+
+Before you start the network make sure you build ercc and ecc_mock. In
+
+```bash
+cd $FPC_PATH/ercc
+make && make docker
+
+cd $FPC_PATH/ecc_mock
+make && make docker
+```
+
+## Prepare network
+
+Go to `$FPC_PATH/integration/fabric-samples` and follow the [instructions](https://hyperledger-fabric.readthedocs.io/en/latest/install.html) to get the fabric binaries and docker images.
+
+```bash
+cd $FPC_PATH/integration/fabric-samples
+curl -sSL https://bit.ly/2ysbOFE | bash -s -- -s
+```
+
+```bash
+cd $FPC_PATH/integration
+./setup.sh
+```
+
+Start network:
+```bash
+cd $FPC_PATH/integration/fabric-samples
+./network.sh up createChannel -c mychannel -ca
```

Comment:
This command fails for me as the docker-compose file manupulation with yq seems to have a bug.
the script fails with 
```
....
rror: failed to create deliver client for orderer: orderer client failed to connect to localhost:7050: failed to create new connection: connection error: desc = "transport: error while dialing: dial tcp [::1]:7050: connect: connection refused"
Channel creation failed
Create channel failed
```
but i think the real cause is a much earlier error message (not causing immediately an abort) which says 
```
...
ERROR: The Compose file './docker/docker-compose-test-net.yaml' is invalid because:
services.peer0.org1.example.com.volumes value ['/var/run/:/host/var/run/', '../organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/msp:/etc/hyperledger/fabric/msp', '../organizations/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/tls:/etc/hyperledger/fabric/tls', 'peer0.org1.example.com:/var/hyperledger/production', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml'] has non-unique elements
services.peer0.org2.example.com.volumes value ['/var/run/:/host/var/run/', '../organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/msp:/etc/hyperledger/fabric/msp', '../organizations/peerOrganizations/org2.example.com/peers/peer0.org2.example.com/tls:/etc/hyperledger/fabric/tls', 'peer0.org2.example.com:/var/hyperledger/production', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0//src/github.com/hyperledger-labs/fabric-private-chaincode/:/opt/gopath/src/github.com/hyperledger-labs/fabric-private-chaincode', '/home/msteiner/src/projects/intel/PoET-et-al/fabric/go-trusted-chain-code.v2.0/src/github.com/hyperledger-labs/fabric-private-chaincode/integration/fabric-samples//config/core.yaml:/etc/hyperledger/fabric/core.yaml'] has non-unique elements

...
```


Actually, figured out the problem: the `yq` scripts are not idempotent for the volume manimpulations for peers! (They are for the orderer).

Reply:
oh true, this is a problem of this script. Added a quick fix